src/common/logger.ts

```diff
@@ -1,35 +1,37 @@
-import * as bunyan from "bunyan"
-const PrettyStream = require("bunyan-prettystream")
+const pino = require("pino")
 const pkg = require("./metadata")
 
-const prettyStdOut = new PrettyStream()
-prettyStdOut.pipe(process.stdout)
+const DEFAULT_LOG_LEVEL = "info"
+const logLevel = (process.env.LOGLEVEL || DEFAULT_LOG_LEVEL).toLowerCase()
+const pactLogFile = process.env.PACT_LOG_PATH
 
-export class Logger extends bunyan {
-  public time(action: string, startTime: number) {
-    const time = Date.now() - startTime
-    this.info(
-      {
-        action,
-        duration: time,
-        type: "TIMER",
-      },
-      `TIMER: ${action} completed in ${time} milliseconds`
-    )
-  }
+const destination = pactLogFile
+  ? pino.destination(pactLogFile)
+  : pino.destination(1)
 
-  public get logLevelName(): string {
-    return bunyan.nameFromLevel[this.level()]
-  }
+const logOpts = {
+  level: logLevel,
+  prettyPrint: {
+    messageFormat: `pact@${pkg.version}: {msg}`,
+    translateTime: true,
+  },
 }
 
-export default new Logger({
-  name: `pact@${pkg.version}`,
-  streams: [
-    {
-      level: (process.env.LOGLEVEL || "info") as bunyan.LogLevel,
-      stream: prettyStdOut,
-      type: "raw",
+let logger = pino(logOpts, destination)
+
+Object.defineProperties(logger, {
```

Comment:
This changes the contract of the `Logger` type from `pino`, since `level` is only a string there, which means calling code might complain about being unable to invoke a function call on a string. I think we might need to export a modified type.

Reply:
Possibly we should export our own type that delegates down to `pino` (or whatever), which would nicely separate the concern, and make any further logging library swaps have less impact.

Reply:
Alternatively we could take advantage of the fact that the logger is a singleton, and export an entirely separate function to update the log level. A quick search of the repo shows only 4 lines (3 files) where this function would be invoked, so that changeset is small.

That avoids needing to define and export a custom type. Thoughts?

Reply:
The function approach sounds very reasonable 👍